src/main/java/org/isf/menu/manager/UserBrowsingManager.java

```diff
@@ -101,40 +97,43 @@ public User newUser(User user) throws OHServiceException {
 		String username = user.getUserName();
 		if (!username.matches(VALID_USERID_PATTERN)) {
 			throw new OHDataValidationException(
-							new OHExceptionMessage(MessageBundle.getMessage("angal.userbrowser.theusernamecontainsinvalidcharacters.msg")));
+				new OHExceptionMessage(MessageBundle.getMessage("angal.userbrowser.theusernamecontainsinvalidcharacters.msg")));
 		}
 		if (ioOperations.isUserNamePresent(username)) {
 			throw new OHDataIntegrityViolationException(
-							new OHExceptionMessage(MessageBundle.formatMessage("angal.userbrowser.theuseralreadyexists.fmt.msg", username)));
+				new OHExceptionMessage(MessageBundle.formatMessage("angal.userbrowser.theuseralreadyexists.fmt.msg", username)));
 		}
 		return ioOperations.newUser(user);
 	}
 
 	/**
 	 * Updates an existing {@link User} in the DB.
-	 *
 	 * @param user - the {@link User} to update
-	 * @return {@code true} if the user has been updated, {@code false} otherwise.
+	 * @return the {@link User} that has been updated
 	 * @throws OHServiceException When failed to update user
 	 */
-	public boolean updateUser(User user) throws OHServiceException {
+	public User updateUser(User user) throws OHServiceException {
+		User oldUser = ioOperations.getUserByNameAndIsDeleted(user.getUserName());
+
+		if (oldUser != null && oldUser.isDeleted() && user.isDeleted()) {
```

Comment:
Isn't getUserByNameAndIsDeleted already returning a isDeleted user (or null)?
If yes, the condition `... && oldUser.isDeleted() ...` is for sure `true`?

Reply:
Nice catch !